### PR TITLE
fixes #2285 - creating subnets - validate length of addresses, strip whitespace and fix common typos

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/models/subnet.rb [hunk 1]
```diff
@@ -17,6 +17,8 @@ class Subnet < ActiveRecord::Base
   validates_uniqueness_of :network
   validates_format_of     :network, :mask,                        :with => Net::Validations::IP_REGEXP
   validates_format_of     :gateway, :dns_primary, :dns_secondary, :with => Net::Validations::IP_REGEXP, :allow_blank => true, :allow_nil => true
+  validates_length_of     :network, :mask, :gateway, :dns_primary, :dns_secondary, :maximum => 15, :message => "must be at most 15 characters"
```
Comment (domcleal) [inline app/models/subnet.rb]: message should be extracted with _()

**Comment by domcleal** [comment]
This doesn't fix the whole issue, try creating a subnet with network address `1.2.3.4` and netmask `255.255.255.0..` and you'll get "invalid address".  The subnet successfully saves and it's on displaying it that it breaks.

It looks like a couple of issues to me, a) IP_REGEXP isn't anchored so leading/trailing noise doesn't match, b) IPAddr.new should be used as a validator too, since it's clearly stricter.

**Comment by isratrade** [comment]
@domcleal, two good points.  re-committed with changes.

**Comment by domcleal** [comment]
@isratrade looks good, thanks.  Would you mind adding a test for IPAddr.new validator too?

**Comment by isratrade** [comment]
@domcleal, I added validation tests, but I could not isolate a test on

``` ruby
validate :ensure_ip_addr_new
```

because this occurs after validates_format  :with => Net::Validations::IP_REGEXP and validates_length_of
Hopefully, the first two validations should catch the error, and if not, IPAddr.new will catch it, but I don't have an example for this edge case.

**Comment by domcleal** [comment]
Ah, good point, I can't think of anything either.  Merged as 31aa5db, thanks @isratrade!

**Pull request closed without merging**